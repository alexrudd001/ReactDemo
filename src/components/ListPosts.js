import React, { useState, useEffect } from 'react';
import Post from './Post';
import ShowMessage from './ShowMessage';

const ListPosts = () => {
    const [ posts, setPosts ] = useState([]);
    const [ messageType, setMessageType ] = useState('');

    useEffect(() => {
        fetch('https://jsonplaceholder.typicode.com/posts')
            .then((res) => {
                if (res.ok) {
                    return res.json();
                }
                throw new Error(res.status);
            })
            .then((responseData) => {
                setPosts(responseData);
            })
            .catch((error) => {
                setMessageType('error');
            });
    }, []);
    return messageType === '' ? posts.length ? (
        posts.map((post) => <Post key={post.id} {...post} />)
    ) : (
        <h3 className="center">Loading posts...</h3>
    ) : (
        <div className="center">
            <ShowMessage messageType={messageType} />
        </div>
    );
};

export default ListPosts;
