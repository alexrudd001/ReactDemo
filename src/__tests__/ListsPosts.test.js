import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListPosts from '../components/ListPosts';

test('renders ListPosts correctly', () => {
    const wrapper = shallow(<ListPosts />);
    expect(toJson(wrapper)).toMatchSnapshot();
});