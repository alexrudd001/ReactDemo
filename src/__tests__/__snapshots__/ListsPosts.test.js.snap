// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListPosts correctly 1`] = `
<h3
  className="center"
>
  Loading posts...
</h3>
`;
